pkg/io/serialize.go

```diff
@@ -50,6 +50,13 @@ func Encode(writer io.Writer, values ...interface{}) (err error) {
 			if err != nil {
 				return errors.WithMessage(err, "marshaling to byte array")
 			}
+
+			var length int64 = int64(len(data))
```

Comment:
Why use `int64`? Use `uint64` or something shorter?

Reply:
Updated to use uint16 in b6137b1cdfaa284311fdb26a48d38c0ce24f1508.